src/features/orders/ordersSlice.ts

```diff
@@ -291,8 +294,17 @@ export const request = createAsyncThunk(
     );
     if (orders.length) {
       const bestOrder = [...orders].sort(orderSortingFunction)[0];
+      const now = Date.now();
       const expiry = parseInt(bestOrder.expiry) * 1000;
-      const timeTilReRequest = expiry - Date.now() - RFQ_EXPIRY_BUFFER_MS;
+      // Due to the sorting in orderSorting function, these orders will be at
+      // the bottom of the list, so if the best one has a very short expiry
+      // so do all the others. Return an empty order array as none are viable.
+      if (expiry - now < RFQ_EXPIRY_BUFFER_MS) return [];
+
+      const timeTilReRequest = Math.min(
```

Comment:
```suggestion
      const timeTilReRequest = Math.max(
```

With `Math.min` `RFQ_MINIMUM_REREQUEST_DELAY_MS` acts like a maximum here.

Reply:
good catch!